collect_app/src/main/java/org/odk/collect/android/widgets/WidgetFactory.java

```diff
@@ -59,14 +59,20 @@ private WidgetFactory() {
     public static QuestionWidget createWidgetFromPrompt(FormEntryPrompt fep, Context context,
                                                         boolean readOnlyOverride) {
 
-        // get appearance hint and clean it up so it is lower case and never null...
+        // Get appearance hint and clean it up so it is lower case and never null.
         String appearance = fep.getAppearanceHint();
         if (appearance == null) {
             appearance = "";
         }
-        // for now, all appearance tags are in english...
+        // For now, all appearance tags are in English.
         appearance = appearance.toLowerCase(Locale.ENGLISH);
 
+        // The search appearance which shows a text area for filtering choices is distinct
+        // from the search() appearance/function. The two can combine but a text area should
+        // not be shown if only the appearance/function is specified.
+        boolean hasSearchAppearance = appearance.contains("search")
```

Comment:
One question: I saw in your test form `appearance="search search('fruits')"` is it also possible to have `appearance="search('fruits') search"` because then I think your approach doesn't work?

Reply:
From http://xlsform.org/en/#dynamic-selects-from-pre-loaded-data: `If the field should use a non-default appearance style. The non-default appearance style goes into the column first, followed by a space, then the search() expression. [e.g., quick search()]`. It's also documented in the code: `search() appearance/function (not part of XForms spec) added by SurveyCTO gets considered in each widget by calls to ExternalDataUtil.getSearchXPathExpression. This means normal appearances should be put before search().` So I think this is sufficient. 

With the change, `appearance="search('fruits') search"` will just mean no search textbox is shown and if a user asks for help we can easily direct them to a fix. 

Does that sound ok?

Reply:
Yes, thanks.